Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,171 @@
+
+Various permission denied errors are avoided by building the Linux kubelet first, per the note at [acs-engine](https://github.com/Azure/acs-engine/blob/master/scripts/build-windows-k8s.sh#L176):
+
+> Due to what appears to be a bug in the Kubernetes Windows build system, one has to first build a Linux binary to generate `_output/bin/deepcopy-gen`. Building to Windows w/o doing this will generate an empty `deepcopy-gen`.
+
+You can optionally execute this in a Vagrant box:
+
+```bash
+DIST_DIR="${HOME}/kube/"
+mkdir ${DIST_DIR}
+SRC_DIR="${HOME}/src/kubernetes/"
+mkdir -p "${SRC_DIR}"
+git clone https://github.com/madhanrm/kubernetes.git ${SRC_DIR}
+cd ${SRC_DIR}
+git checkout cniwindows
+build/run.sh make WHAT=cmd/kubelet KUBE_BUILD_PLATFORMS=linux/amd64
+build/run.sh make WHAT=cmd/kubelet KUBE_BUILD_PLATFORMS=windows/amd64
+cp ${SRC_DIR}/_output/dockerized/bin/windows/amd64/kubelet.exe ${DIST_DIR}
+#winkernelproxy has already been merged but still needs to be built
+git checkout winkernelproxy 
+build/run.sh make WHAT=cmd/kube-proxy KUBE_BUILD_PLATFORMS=windows/amd64
+cp ${SRC_DIR}/_output/dockerized/bin/windows/amd64/kube-proxy.exe ${DIST_DIR}
+ls ${DIST_DIR}
+```
+
+Done! Now, we also need the actual Linux Kubernetes binaries for v1.8. You can pull these from [the Kubernetes mainline](https://github.com/kubernetes/kubernetes/releases/tag/v1.8.0), or build them from source as above, except using `K8SREPO=k8s.io/kubernetes` and the `release-1.8` branch, stopping before the `export ...` line.
+
+If you built them from source, copy the binaries (at least `hyperkube`, `kubectl`, and `kubeadm`) directly to `~/kube/bin`. Otherwise, you will need to extract the downloaded archive and run the `cluster/get-kube-binaries.sh` script before copying the binaries.
+
+Copy these to `~/kube/bin` (note the extra `/bin` compared to the above path; we actual need to use these during master configuration, rather than just copying them to the Windows node). The v1.8 Windows binary `kubectl.exe` is included in the `windows/` directory of this repo, but you can also build it here (setting `KUBE_BUILD_PLATFORMS=windows/amd64` and `WHAT=cmd/kubectl` as above).
+
+### Install CNI Plugins ###
+
+We also need to install the basic CNI plugin binaries so that networking works. Download them from [here](https://github.com/containernetworking/plugins/releases) and copy them to `/opt/cni/bin/`.
+
+```bash
+DOWNLOAD_DIR="${HOME}/kube/cni-plugins"
+CNI_BIN="/opt/cni/bin/"
+mkdir ${DOWNLOAD_DIR}
+cd $DOWNLOAD_DIR
+curl -L $(curl -s https://api.github.com/repos/containernetworking/plugins/releases/latest | grep browser_download_url | grep 'amd64.*tgz' | head -n 1 | cut -d '"' -f 4) -o cni-plugins-amd64.tgz

```

Comment:
Requires curl, do you want to ask them to apt install this earlier on ?

Reply:
Is curl really not included in almost every distro? :) added!